backend/app/store/service/service.go

```diff
@@ -855,6 +874,61 @@ func (s *DataStore) Last(siteID string, limit int, since time.Time, user store.U
 	return s.alterComments(comments, user), nil
 }
 
+// Search commens using user query
+func (s *DataStore) Search(siteID, query, sortBy string, from, limit int) (*SearchResultPage, error) {
```

Comment:
Part up to there (server.go -> rest_public.go -> service.go, engine changes) seems sane, I have no questions about the flow of data.
At this point I haven't reviewed only the new `search` module.

Reply:
UPD: except for `WrapEngine`.